Model/Service/Index.php

```diff
@@ -358,11 +359,26 @@ public function markAsInSync(ProductIndexInterface $productIndex)
         }
     }
 
+    /**
+     * Defines product index item as dirty
+     *
+     * @param ProductIndexInterface $productIndex
+     * @throws Exception
+     * @return void
+     */
+    public function markAsDirty(ProductIndexInterface $productIndex)
```

Comment:
Can be removed after `markCurrentItemsAsDirtyByStore` is implement in the collection